docs/sphinx/using/cudaq/platform.rst

```diff
@@ -17,37 +17,116 @@ specific asynchronous function invocations targeting a desired QPU.
+
+The results might look like the following (4 different random samplings).
+
+.. code-block:: console 
+  
+    Number of QPUs: 4
+    { 10011:28 01100:28 ... }
+    { 10011:37 01100:25 ... }
+    { 10011:29 01100:25 ... }
+    { 10011:33 01100:30 ... }
+
+.. note:: 
+
+  By default, the :code:`nvidia-mqpu` platform will utilize all available GPUs (number of QPUs instances is equal to the number of GPUs).
+  To specify the number QPUs to be instantiated, one can set the :code:`CUDAQ_MQPU_NGPUS` environment variable.
+  For example, :code:`export CUDAQ_MQPU_NGPUS=2` to specify that only 2 QPUs (GPUs) are needed.
+
+
+An equivalent example in Python is as follows.
+
+.. literalinclude:: ../../snippets/python/using/cudaq/platform/sample_async.py
+    :language: python
+    :lines: 8-
+
+Asynchronous expectation value computations
++++++++++++++++++++++++++++++++++++++++++++
+
+One typical use case of the :code:`nvidia-mqpu` platform is to distribute the 
+expectation value computations of a multi-term Hamiltonian across multiple virtual QPUs (:code:`GPUEmulatedQPU`).
+
+Here is an example.
+
+.. literalinclude:: ../../snippets/cpp/using/cudaq/platform/observe_mqpu.cpp
+    :language: cpp
+    :lines: 17-36
+
+
+One can then target the :code:`nvidia-mqpu` platform by:
+
+.. code-block:: console 
+
+    nvq++ observe_mqpu.cpp -target nvidia-mqpu
+    ./a.out
+
+Equivalently, in Python
+
+.. literalinclude:: ../../snippets/python/using/cudaq/platform/observe_mqpu.py
+    :language: python
+    :lines: 8-
+
+In the above code snippet, since the Hamiltonian contains four non-identity terms, there are four quantum circuits that need to be executed
+in order to compute the expectation value of that Hamiltonian and given the quantum state prepared by the ansatz kernel. When the :code:`nvidia-mqpu` platform
+is selected, these circuits will be distributed across all available QPUs. The final expectation value result is computed from all QPU execution results.
+
+Parallel distribution mode
+++++++++++++++++++++++++++
+
+The CUDA Quantum :code:`nvidia-mqpu` platform supports two modes of parallel distribution of expectation value computation:
+
+* MPI: distribute the expectation value computations across available MPI ranks and GPUs for each Hamiltonian term.
+* Thread: distribute the expectation value computations among available GPUs via standard C++ threads (each thread handles one GPU).
+
+For instance, if all GPUs are available on a single node, thread-based parallel distribution 
+(:code:`cudaq::parallel::thread` in C++ or :code:`cudaq.parallel.thread` in Python, as shown in the above example) is sufficient. 
+On the other hand, if one wants to distribute the tasks across GPUs on multiple nodes, e.g., on a compute cluster, MPI distribution mode
+should be used.
+
+An example of MPI distribution mode usage is as follows:
+
+C++
+^^^
+
+.. literalinclude:: ../../snippets/cpp/using/cudaq/platform/observe_mqpu_mpi.cpp
+    :language: cpp
+    :lines: 17-32
```

Comment:
It might be helpful to use the `:start-after:` and `:end-before` attributes instead of calling out specific line numbers here. If somebody changes those files in the future, the line numbers will probably change, and it might not be very obvious that docs look incorrect at that point. See [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude) for an example. (I have not tested this.)